langchain/document_loaders/base.py

```diff
@@ -3,11 +3,11 @@
 from typing import Iterator, List, Optional
 
 from langchain.document_loaders.blob_loaders import Blob
-from langchain.schema import Document
+from langchain.schema import BaseRetriever, Document
 from langchain.text_splitter import RecursiveCharacterTextSplitter, TextSplitter
 
 
-class BaseLoader(ABC):
+class BaseLoader(BaseRetriever):
```

Comment:
This shows one way (which I don't love) to deal with the fact that config information must be supplied at initialization https://github.com/hwchase17/langchain/pull/4731/files

Anyway we may revisit this since I agree that being unable to provide config through the method leads to some akward downstream patterns